event_buffer.go

```diff
@@ -0,0 +1,71 @@
+package statful
+
+import (
+	"bytes"
+	"encoding/json"
+	"sync"
+)
+
+type eventBuffer struct {
+	buffer     []Event
+	dryRun     bool
+	eventCount int
+	flushSize  int
+	mu         sync.Mutex
+
+	Logger Logger
+	Sender Sender
+}
+
+func (e *eventBuffer) Event(event Event) {
+	e.mu.Lock()
+
+	e.buffer = append(e.buffer, event)
+	e.eventCount++
+
+	e.mu.Unlock()
+}
+
+func (e *eventBuffer) Flush() error {
+	e.mu.Lock()
+
+	events := e.drainBuffers()
+	e.mu.Unlock()
+
+	return e.flushBuffers(events)
+}
+
+func (e *eventBuffer) flushBuffers(buffer []Event) error {
+	var err error
+
+	if len(buffer) > 0 {
+		if e.dryRun {
+			for _, event := range buffer {
+				e.Logger.Println("Dry event: ", event)
```

Comment:
This log is going to be a bit weird when it comes to formatting I think (no field names etc.). Maybe we should implement the `Stringer` interface on Event if we want to log it a little bit nicer. 

Not super important tho since this is only for dry runs.